src/main/java/com/example/bookstoreweb/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,113 @@
+import com.example.bookstoreweb.model.Order;
+import com.example.bookstoreweb.model.OrderItem;
+import com.example.bookstoreweb.model.ShoppingCart;
+import com.example.bookstoreweb.model.User;
+import com.example.bookstoreweb.repository.OrderItemRepository;
+import com.example.bookstoreweb.repository.OrderRepository;
+import com.example.bookstoreweb.repository.ShoppingCartRepository;
+import com.example.bookstoreweb.repository.UserRepository;
+import com.example.bookstoreweb.service.OrderService;
+import com.example.bookstoreweb.service.UserService;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Objects;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final UserService userService;
+    private final UserRepository userRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+
+    @Override
+    public List<OrderResponseDto> findAllOrdersByUser(String email, Pageable pageable) {
+        UserResponseDto responseDto = userService.getByEmail(email);
+        return orderRepository.findAllByUserId(responseDto.getId()).stream()
+                .map(orderMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public OrderResponseDto create(String email, OrderRequestDto requestDto) {
+        User user = userRepository.findByEmail(email).orElseThrow(
+                () -> new RuntimeException("Can't find user by email:" + email)
+        );
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(user.getId())
+                .orElseThrow(() -> new RuntimeException("Can't find shopping cart"));
+
+        Order order = orderMapper.toModel(requestDto);
+        order.setUser(user);
+        order.setStatus(Order.Status.PENDING);
+        order.setTotal(shoppingCart.getCartItems().stream()
+                .map(cartItem -> cartItem.getBook().getPrice()
+                        .multiply(BigDecimal.valueOf(cartItem.getQuantity())))
+                .reduce(BigDecimal.ZERO, BigDecimal::add));
+        order.setOrderDate(LocalDateTime.now());
+        Order save = orderRepository.save(order);
+        for (CartItem cartItem : shoppingCart.getCartItems()) {
+            OrderItem orderItem = new OrderItem();
+            orderItem.setBook(cartItem.getBook());
+            orderItem.setOrder(save);
+            orderItem.setPrice(cartItem.getBook().getPrice());
+            orderItem.setQuantity(cartItem.getQuantity());
+            order.getOrderItems().add(orderItem);
+            orderItemRepository.save(orderItem);
+        }
+        return orderMapper.toDto(save);
```

Comment:
1) Verbs are for methods naming. Variable with name "save" confuses, could be savedOrder or smth similar
2) Why we need separate variable? Can use the one we have
3) You have mapstruct, create mapper for CartItem -> OrderItem conversion. Or create constructor for OrderItem that accepts CartItem as argument
4) same for Order, it seems logical to have constructor from shopping cart as it's natural process that cart becomes an order

then would look like below or similar
```suggestion
        Order order = new Order(shoppingCart);
        order.setShippingAddress(requestDto.getShippingAddress());
        orderRepository.save(order);
        for (CartItem cartItem : shoppingCart.getCartItems()) {
            OrderItem orderItem = new OrderItem(cartItem);
            orderItem.setOrder(order);
            order.getOrderItems().add(orderItem);
            orderItemRepository.save(orderItem);
        }
        return orderMapper.toDto(order);
```